apps/src/code-studio/pd/foorm/FoormEditor.jsx

```diff
@@ -193,33 +199,187 @@ class FoormEditor extends React.Component {
+                  onChange={e => this.setState({day: e.target.value})}
+                />
+              </label>
+              <label>
+                workshop_agenda <br />
+                <input
+                  type="text"
+                  value={this.state.workshop_agenda}
+                  onChange={e =>
+                    this.setState({workshop_agenda: e.target.value})
+                  }
+                />
+              </label>
+            </form>
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  renderPreview() {
+    return (
+      <div className="foorm-preview">
+        <div style={styles.previewBox}>
+          {this.props.formHasError && (
+            <div style={styles.errorMessage}>
+              <FontAwesome icon="exclamation-triangle" /> There is a parsing
+              error in the survey configuration. Errors are noted on the left
+              side of the editor.
+            </div>
+          )}
+          {this.state.libraryError && (
+            <div style={styles.errorMessage}>
+              <FontAwesome icon="exclamation-triangle" />
+              {`There is an error in the use of at least one library question. The error is: ${
+                this.state.libraryErrorMessage
+              }`}
+            </div>
+          )}
+          {this.state.formPreviewQuestions &&
+            !this.props.formHasError &&
+            !this.state.libraryError && (
+              // key allows us to force re-render when preview is called
+              <Foorm
+                formQuestions={this.state.formPreviewQuestions}
+                formName={'preview'}
+                formVersion={0}
+                submitApi={'/none'}
+                key={`form-${this.state.formKey}`}
+                surveyData={{
+                  facilitators: this.state.facilitators,
+                  num_facilitators: this.state.num_facilitators,
+                  workshop_course: this.state.workshop_course,
+                  workshop_subject: this.state.workshop_subject,
+                  regional_partner_name: this.state.regional_partner_name,
+                  is_virtual: this.state.is_virtual,
+                  day: this.state.day,
+                  is_friday_institute: this.state.is_friday_institute,
+                  workshop_agenda: this.state.workshop_agenda
+                }}
+              />
+            )}
+        </div>
+      </div>
+    );
+  }
+
   render() {
     return (
-      <div style={styles.foormEditor}>
-        <div style={styles.editorAndOptions}>
+      <div>
+        {this.props.formName && (
+          <div style={styles.editorHeader}>
+            <h2>
+              {`${this.props.formName}, version ${this.props.formVersion}`}
+              <br />
+              {`Form State: ${
+                this.props.isFormPublished ? 'Published' : 'Draft'
+              }`}
+            </h2>
+          </div>
+        )}
+        <div style={styles.livePreview}>
```

Comment:
Moving this up to the top was a great move! It will make it so much more accessible for editors